sheriff.go

```diff
@@ -51,6 +54,12 @@ func Marshal(options *Options, data interface{}) (interface{}, error) {
 	v := reflect.ValueOf(data)
 	t := v.Type()
 
+	// Initialise nestedGroupsMap,
+	// TODO: this may impact the performance, find a better place for this.
+	if options.nestedGroupsMap == nil {
```

Comment:
we could probably move to a `NewOptions` (or similar) function which initializes the `Options` struct for that.

Unsure if needed, so leave it as it is for the moment.